subsys/net/l2/ethernet/ethernet.c

```diff
@@ -374,11 +336,30 @@ static enum net_verdict ethernet_recv(struct net_if *iface,
 			       net_pkt_lladdr_dst(pkt));
 	}
 
+	STRUCT_SECTION_FOREACH(net_l3_register, l3) {
+		bool is_addr_ok;
+
+		if (l3->ptype != type || l3->l2_addr_check == NULL ||
+		    net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
```

Comment:
I'm a bit puzzled with `net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)` check, it seems redundant, this is literally Ethernet L2 implementation, how could a packet end up here if it's not received on an Ethernet interface? Or am I missing something here?

Reply:
Yes, good point, the check is clearly wrong. What I was trying to check here, is to make sure that if there are multiple network interfaces with Ethernet and some other L2 type, and if both have added `NET_L3_REGISTER()` calls, we should only check Ethernet ones here. Need to think how to accomplish this.